src/backend/libs/efiling-bambora-api-client/src/main/java/ca/bc/gov/open/jag/bambora/BamboraPaymentAdapter.java

```diff
@@ -41,7 +41,7 @@ public PaymentTransaction makePayment(EfilingPayment efilingPayment) {
             result.setInvoiceNo(response.getOrderNumber());
             result.setTransactonDtm(DateTime.now());
             result.setTransactionAmt(BigDecimal.valueOf(response.getAmount()));
-            if (response.getApproved() == PaymentConstants.BAMBORA_APPROVAL_RESPONSE) {
+            if (response.getApproved().equals(PaymentConstants.BAMBORA_APPROVAL_RESPONSE)) {
```

Comment:
This is definitely better, however, both are examples of autoboxing/unboxing.
The left operand is an Object while the right is a primitive int.
It's just my opinion, but I was always taught to never rely on Java's autoboxing feature.  It can be easier to write code, but can also introduce extremely difficult to track down bugs.  It can be considered a best practice to set your IDE to show autoboxing as compiler errors.

Here are some examples of hard-to-find bugs due to autoboxing:
https://dzone.com/articles/autoboxing-and-its-pitfalls-1
https://andrew-flower.com/blog/Java-Unboxing-Pitfall